encord/http/querier.py

```diff
@@ -157,10 +149,15 @@ def _do_request(req, timeouts: tuple):
 
         return res_json.get("response")
 
-    @staticmethod
-    def create_new_session() -> Session:
-        session = Session()
 
-        session.mount("https://", HTTPAdapter(max_retries=Retry(connect=0)))
+@contextmanager
+def create_new_session(max_retries: Optional[int] = None) -> Session:
+    retry_policy = Retry(total=None, connect=max_retries, read=max_retries)
+
+    with Session() as session:
+        http_adapter = HTTPAdapter(max_retries=retry_policy)
+
+        session.mount("http://", http_adapter)
```

Comment:
That should not be needed right? Or do we have any http outgoing requests?

Reply:
Well, you can use the SDK against host not-authenticated with http. Such as when running tests locally or against internal environments, or on-prem deployments without a certificate.

If we don't configure it here, it just gets defaulted to an HTTPAdapter()